impc_etl/jobs/parse/gene_production_status_parser.py

```diff
@@ -0,0 +1,196 @@
+import logging
+import textwrap
+from typing import Dict, List
+
+from airflow.sdk import Variable, asset
+from pyspark.sql.functions import (
+    when, col, lit, lower, udf, collect_set, array_contains
+)
+from pyspark.sql.types import StringType, IntegerType
+
+from impc_etl.utils import utils
+from impc_etl.utils.airflow import create_input_asset, create_output_asset
+from impc_etl.utils.spark import with_spark_session
+
+logger = logging.getLogger("airflow.task")
+
+dr_tag = Variable.get("data_release_tag")
+
+gentar_gene_interest_asset = create_input_asset("tracking/gene_interest.tsv")
+product_report_parquet_asset = create_input_asset("output/product_report_parquet")
+gene_status_output_asset = create_output_asset("gene_status_parquet")
+
+@asset.multi(
+    schedule=[gentar_gene_interest_asset, product_report_parquet_asset],
+    outlets=[gene_status_output_asset],
+    dag_id=f"{dr_tag}_gene_production_status_parser",
+    description=textwrap.dedent(
+        """
+        Parses gene production and phenotyping statuses by integrating gene interest data from Gentar
```

Comment:
Wherever is possible include URLS to the original datasource, check the doc strign from the original Luigi task as example:

```
    """
    PySpark Task class to parse gene status information from tracking systems reports.
    This also maps the different status in the reports to the ones we use on the website.
    It takes as an input the Gene Interest report from gentar (https://www.gentar.org/tracker-api/api/reports/gene_interest)
    And the Products report from Gentar ().
    The output is a Parquet file containing a full report of the different Statuses available for a Gene.
    """
```